tests/baselines/reference/assignObjectToNonPrimitive.js

```diff
@@ -4,6 +4,11 @@ var y = {foo: "bar"};
 var a: object;
 a = x;
 a = y;
+
+// assignable to right hand side of forIn
+for (var key in a) {
+    var val = a[key]
+}
```

Comment:
This test (`conformance/types/nonPrimitive/assignObjectToNonPrimitive.ts`) seems unrelated - would it be possible to remove this and create a new test? Something like `conformance/types/nonPrimitive/indexingWithForInLoop01.ts` seems more appropriate.